federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -728,8 +728,12 @@ func (s *ServiceController) lockedUpdateDNSRecords(service *cachedService, clust
 	for key := range s.clusterCache.clientMap {
 		for _, clusterName := range clusterNames {
 			if key == clusterName {
-				s.ensureDnsRecords(clusterName, service)
-				ensuredCount += 1
+				err := s.ensureDnsRecords(clusterName, service)
+				if err != nil {
+					glog.Errorf("Failed to ensure DNS records for service %v in cluster %s due to %v", service, clusterName, err)
+				} else {
+					ensuredCount += 1
+				}
 			}
 		}
 	}
```

Comment:
Reasonable. The numbers may be counted by below?
1. number of missing cluster clients: count of `key != clusterName`
2. number of failed ensureDNsRecords: count of `when s.ensureDnsRecords return non nil `?

If so, we can define two new variables: `missedCount`, `unensuredCount`(instead current `ensuredCount`)